batch.go

```diff
@@ -0,0 +1,126 @@
+package ed25519consensus
+
+import (
+	"crypto/ed25519"
+	"crypto/rand"
+	"crypto/sha512"
+
+	"filippo.io/edwards25519"
+)
+
+// Verifier holds entries of public keys, signature and a scalar which are used for batch verification.
+type Verifier struct {
+	entries   []ks
+	batchSize uint32
```

Comment:
In the `ed25519-dalek` implementation, there's a separate batch size variable because the batch entries are stored in a map from public keys to batch entries, so that the implementation can automatically coalesce terms from the same public key.  This implementation doesn't do that optimization (which is fine -- doing batch verification is already a big speedup, and if there were a good use case for it, this API allows it to be done later), so does it need to track `batchSize` separately from `len(entries)`?

Reply:
Changed it over to len(entries). I'm not sure how much more of a speedup that will lend, but this is already 100%-150% faster than our current system.